### DISCUSSION TITLE
Reported CVE URLs go to 404

### DISCUSSION THREAD

**Discussion by emcfins** [aquasecurity/trivy#7487] [Documentation]
### Description

We run reports and provide links to the dev teams we support to any found CVEs when running through Trivy. This includes the aqcuasec links so the teams can learn more. 
We have found that many of the links go to 404. For example:

- "PrimaryURL": "https://avd.aquasec.com/nvd/cve-2024-43800"
- "PrimaryURL": "https://avd.aquasec.com/nvd/cve-2024-43788",
- "PrimaryURL": "https://avd.aquasec.com/nvd/cve-2024-43796",
- "PrimaryURL": "https://avd.aquasec.com/nvd/cve-2024-43788",

All 4 of those appear in a report and they all go to 404 pages.

### Suggestions

In what circumstance does this happen? Is there a way you can not remove them from your database website?

Or - if they're being reviewed, redirect the user to a page that explains the current status of that CVE?

**Comment by knqyf263** [comment]
@simar7 @DmitriyLewen Isn't AVD working now?

**Reply by simar7** [reply]
Looking at the action history it seems there's no problem with AVD https://github.com/aquasecurity/avd-generator/actions 

Does the advisory exist in vuln-list?

**Comment by simar7** [comment]
It looks like these vulnerabilities were just added yesterday https://github.com/aquasecurity/vuln-list-nvd/commit/ba9c0366529a48c711aafcfadeea1bb48e6b449a

And the AVD generator was just kicked off today (happens once a day). They seem to be live now on AVD.

We could run AVD more often than vuln-list updates, that way we wouldn't miss any. @knqyf263 what do you think?

**Reply by knqyf263** [reply]
>We could run AVD more often than vuln-list updates, that way we wouldn't miss any. @knqyf263 what do you think?

Yes, it would be great, but if I remember correctly, the workflow is heavy, and we use macOS runner. Running more often costs more?

**Reply by simar7** [reply]
Actually we've switched over to using Linux runners now and the time has been shortened as well (average 20-30min per regular run). So we can run it more often. I'll create a PR to improve it.

**Reply by simar7** [reply]
PR for review https://github.com/aquasecurity/avd-generator/pull/97

**Reply by knqyf263** [reply]
Nice. A few days ago, I also saw someone facing the 404 problem in AVD. More frequent updates will help.

**Reply by itaysk** [reply]
sorry i forgot to update here but after seeing this i checked the generator Action and saw it was disabled due to inactivity in the repo. I released it and kicked off a manual build which added these recent vulns

**Reply by simar7** [reply]
Looks like we can use a keep alive action like this [one](https://github.com/gautamkrishnar/keepalive-workflow) PR here: https://github.com/aquasecurity/avd-generator/pull/98

**Reply by knqyf263** [reply]
Looks like the action uses [this API](https://docs.github.com/en/rest/actions/workflows?apiVersion=2022-11-28#enable-a-workflow). 

Adding `gh api --method PUT /repos/${{ github.repository }}/actions/workflows/${{ github.workflow }}/enable` seems enough (I've not debugged this one; `WORKFLOW_ID` might be wrong).
